app/src/main/java/doit/study/droid/QuestionFragment.java

```diff
@@ -44,16 +46,19 @@
     private LinearLayout mvAnswersLayout;
     private TextView mvRight;
     private TextView mvWrong;
+    private Button mvNextButton;
 
 
     // Factory method
-    public static QuestionFragment newInstance(int position) {
+    public static QuestionFragment newInstance(int position, int questionId, AnswerCheckListener listener) {
         if (DEBUG) Log.i("NSA", "newInstance "+position);
         // add Bundle args if needed here before returning new instance of this class
         QuestionFragment fragment = new QuestionFragment();
         Bundle bundle = new Bundle();
         bundle.putInt(ID_KEY, position);
         fragment.setArguments(bundle);
+        fragment.mQuestionId = questionId;
+        fragment.mAnswerCheckListener = listener;
```

Comment:
What about having two buttons "commit" and "next" visible at the same time?
If user clicks next without committing, it means that he skipped the question and will return to it later (after the last question in the set).
If user answers wrong, he has two options: either try one more time or move to the next question.
But if user can try one question more then one times, doesn't it break the statistics? If user doesn't know the right answer and tries to guess it, he will make his statistics worse and worse with every commit. I need to understand the statistics logic you are implementing.